core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -527,7 +531,9 @@ static NameResolver getNameResolver(String target, NameResolver.Factory nameReso
         // Should not be possible.
         throw new IllegalArgumentException(e);
       }
-      NameResolver resolver = nameResolverFactory.newNameResolver(targetUri, nameResolverParams);
+      NameResolver resolver = nameResolverFactory.newNameResolver(
+          targetUri, nameResolverParams,
+          proxyDetector.proxyFor(GrpcUtil.unresolvedAddress(targetUri, defaultPort)) != null);
```

Comment:
The URIs are NameResolver-specific. `ManagedChannelImpl` can't understand them; it relies entirely on the NameResolver. So this should go back the way it was.

I actually don't quite follow why it was necessary. I would have thought the ManagedChannelBuilder change would have been enough.